libs/GraphicECS/SFML/Systems/InputManagement.cpp

```diff
@@ -70,6 +73,33 @@ namespace graphicECS::SFML::Systems
         }
     }
 
+    void InputManagement::_textEnteredEvents(sf::Event &event, std::vector<std::shared_ptr<Entity>> joined)
+    {
+        if (event.type == sf::Event::TextEntered) {
+            auto writeText = [&event](std::shared_ptr<Entity> entityPtr) {
+                auto guard = std::lock_guard(*entityPtr.get());
```

Comment:
You should use the lock_guard each time you want to update some value, avoid this usage at the beginning of a lambda in this exemple

Reply:
:eyes: